src/test/java/com/linkedin/kafka/clients/utils/DefaultHeaderSerializerDeserializerTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.utils;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.util.Arrays;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class DefaultHeaderSerializerDeserializerTest {
+  private static final char UNICODE_REPLACEMENT_CHARACTER = 65533;
+
+  @Test
+  public void invalidUtf8Magic() throws Exception {
```

Comment:
We spent time making the magic number can be decoded as a valid UTF-8 string.  This tests that it is not a valid UTF-8 string.